sherpa/tests/test_fit_unit.py

```diff
@@ -2826,3 +2826,252 @@ def test_fit_ensures_data_and_model_dimensionality_matches(data_class, ddim, mod
     with pytest.raises(DataErr,
                        match=f"Data and model dimensionality do not match: {ddim}D and {mdim}D"):
         Fit(data, model)
+
+
+def check_str(out, expecteds):
+    """Check that out, when split on a newline, matches expecteds"""
+
+    toks = str(out).split("\n")
+    for tok, expected in zip(toks, expecteds):
+        assert tok == expected
+
+    assert len(toks) == len(expecteds)
+
+
+def test_fit_results_str():
+    """Just check we so call str() on a fit results instance"""
+
+    data = make_data(Data1D)
+    model = Const1D()
+    fit = Fit(data, model, stat=Cash())
+    fres = fit.fit()
+    assert fres.succeeded
+
+    check_str(fres,
+              ["datasets       = None",
+               "itermethodname = none",
+               "methodname     = levmar",
+               "statname       = cash",
+               "succeeded      = True",
+               "parnames       = ('const1d.c0',)",
+               "parvals        = (3.33332926058462,)",
+               "statval        = -4.079456086503793",
+               "istatval       = 6.0",
+               "dstatval       = 10.079456086503793",
+               "numpoints      = 3",
+               "dof            = 2",
+               "qval           = None",
+               "rstat          = None",
+               "message        = successful termination",
+               "nfev           = 9"])
+
+
+def test_fit_itermethod_requires_name():
+    """Just check we error out gracefully.
+
+    This is a corner case.
+    """
+
+    data = make_data(Data1D)
+    model = Const1D()
+    with pytest.raises(ValueError,
+                       match="^Missing name field in itermethod_opts argument$"):
+        Fit(data, model, itermethod_opts={"grow": 2})
+
+
+def test_fit_itermethod_requires_known_name():
+    """Just check we error out gracefully.
+
+    This is a corner case.
+    """
+
+    data = make_data(Data1D)
+    model = Const1D()
+    with pytest.raises(ValueError,
+                       match="^not-a-name is not an iterative fitting method$"):
+        Fit(data, model, itermethod_opts={"name": "not-a-name", "grow": 2})
+
+
+def test_fit_results_with_iteration_str():
+    """Just check we so call str() on a fit results instance with an iteration method
+
+    We have not picked a dataset where the
```

Comment:
```suggestion
    """Just check we can call str() on a fit results instance with an iteration method
    """
```

Reply:
I'd originally meant to add a point that the database wasn't set up for actually causing the sigmarej outine to fire, hence the incomplete sentence which you've removed, but I went and created something that did actually have an outlier just to make sure, so a double reason to remove the text...